### PR TITLE
CHET-435: deploy standalone or with vpc

### PR DESCRIPTION
**Pull request by bpar476**
## Summary

* Implement ability to deploy with-vpc or standalone quickstart in backend
* Expose configurable deployment via API
* Call API with deployment mode depending on user choice in ASI configuration screen

## Note

* Currently, the JSON is not deserializing nicely into the `ProvisioningConfig` expected by the backend. Will probably have to work on this next week.

### CODE REVIEW

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/infrastructure/QuickstartDeploymentService.java [hunk 1]
```diff
@@ -42,9 +43,11 @@
     static final String SECURITY_GROUP_NAME_STACK_OUTPUT_KEY = "SGname";
 
     private final Logger logger = LoggerFactory.getLogger(QuickstartDeploymentService.class);
-    private static final String QUICKSTART_TEMPLATE_URL = "https://trebuchet-public-resources.s3.amazonaws.com/quickstart-jira-dc-with-vpc.template.yaml";
+    private static final String QUICKSTART_WITH_VPC_TEMPLATE_URL = "https://trebuchet-public-resources.s3.amazonaws.com/quickstart-jira-dc-with-vpc.template.yaml";
```
Comment (dylanrathbone) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/infrastructure/QuickstartDeploymentService.java:46]: Might be nice to turn the root context into a constant too i.e. `https://trebuchet-public-resources.s3.amazonaws.com/` and then interpolate the target string with it.

File: spi/src/test/kotlin/com/atlassian/migration/datacenter/spi/infrastructure/ProvisioningConfigTest.kt [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
+        assertEquals("iamsupersecure.trustme", params["password"])
+        assertEquals("2", params["instanceCount"])
+        assertEquals("false", params["multiAZ"])
+    }
+
+    @Test
+    fun shouldBeDeserialisable() {
+        val jiraProductJSONKey = "JiraProduct"
+        val jiraProductJSONValue = "Software"
+        val stackName = "bpartridge-treb-20"
```
Comment (dylanrathbone) [inline spi/src/test/kotlin/com/atlassian/migration/datacenter/spi/infrastructure/ProvisioningConfigTest.kt:29]: you sure you want to commit your username here?

Comment (bpar476) [reply spi/src/test/kotlin/com/atlassian/migration/datacenter/spi/infrastructure/ProvisioningConfigTest.kt:29]: Probably not a big deal, but I've changed it :)

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/infrastructure/QuickstartDeploymentService.java (hunk 1 above)
Comment (bpar476) [reply core/src/main/java/com/atlassian/migration/datacenter/core/aws/infrastructure/QuickstartDeploymentService.java:46]: Good suggestions, done :)

**Pull request merged**